core/pkg/ingress/types.go

```diff
@@ -40,6 +41,10 @@ var (
 // Controller holds the methods to handle an Ingress backend
 // TODO (#18): Make sure this is sufficiently supportive of other backends.
 type Controller interface {
+	// HealthzChecker returns is a named healthz check that returns the ingress
+	// controller status
+	healthz.HealthzChecker
```

Comment:
> What if I want to ps aux | grep nginx as a health check in my backend?

you can do that inside the Check method of you backend.
Just in case no matter what/how you backend is checking the yaml always remains like this:
```
        readinessProbe:
          httpGet:
            path: /healthz
            port: 10254
            scheme: HTTP
        livenessProbe:
          httpGet:
            path: /healthz
            port: 10254
            scheme: HTTP
```